src/Tabs/TabGroup.js

```diff
@@ -0,0 +1,111 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import { TabContent } from './_TabContent';
+import React, { Component } from 'react';
+
+export class TabGroup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedId: props.selectedId
+        };
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        return nextProps.selectedId !== prevState.selectedId ? { selectedId: nextProps.selectedId } : null;
+    }
+
+    // set selected tab
+    handleTabSelection = (event, id) => {
+        event.preventDefault();
+        this.setState({
+            selectedId: id
+        });
+
+        if (this.props.onTabClick) {
+            this.props.onTabClick(event, id);
+        }
+    };
+
+    // clone Tab element
+    cloneElement = (child) => {
+        return (React.cloneElement(child, {
+            onClick: this.handleTabSelection,
+            selected: this.state.selectedId === child.props.id
+        }));
+    }
+
+    // create tab list
+    renderTabs = () => {
+        return React.Children.map(this.props.children, (child) => {
+            return (
+                <li
+                    {...this.props.tabLinkProps}
+                    className='fd-tabs__item'
+                    key={child.props.id}>
+                    {this.cloneElement(child)}
+                </li>);
```

Comment:
I _think_ it might be better to have `Tab` render both the `<li>` and the `<a>` elements for each tab.  I just think keeps "like" elements together.  Otherwise, `Tab` looks like a misnamed link renderer.  The map can stay here and then `className` and `key` could be moved to `Tab` as well.